src/bundles.ts

```diff
@@ -87,46 +86,40 @@ interface UpdateRemoteBundleOptions extends ConnectionOptions {
  * @param sessionToken
  * @param remoteBundle
  */
-export async function uploadRemoteBundle({
-  maxPayload = MAX_PAYLOAD,
-  ...options
-}: UpdateRemoteBundleOptions): Promise<boolean> {
+export async function uploadRemoteBundle(options: UpdateRemoteBundleOptions): Promise<void> {
   let uploadedFiles = 0;
   emitter.uploadBundleProgress(0, options.files.length);
 
   const apiParams = pick(options, ['baseURL', 'sessionToken', 'source', 'bundleHash']);
 
-  const uploadFileChunks = async (bucketFiles: FileInfo[]): Promise<boolean> => {
-    const resp = await extendBundle({
+  const uploadFileChunks = async (bucketFiles: FileInfo[]): Promise<void> => {
+    // Note: we specifically create __new__ isolated bundles here to faster files upload
+    const resp = await createBundle({
       ...apiParams,
-      files: bucketFiles.reduce((d, f) => ({ ...d, [f.bundlePath]: pick(f, ['hash', 'content']) }), {}),
+      files: bucketFiles.reduce((d, f) => {
+        d[f.bundlePath] = pick(f, ['hash', 'content']);
+        return d;
+      }, {}),
     });
 
-    // During upload process, we expect the bundleHash not to change (same file map)
-    if (resp.type !== 'error' && resp.value.bundleHash === apiParams.bundleHash) {
```

Comment:
I don't agree with this change (and all of its consequences, like the removal of the tests and the final bundle check). I added this check specifically because we had cases where the upload request was creating a new bundle and we don't take the returning `resp.value.bundleHash` into consideration in future steps. We currently use the ExtendBundle API both for extending the bundle definition (list of files) and to upload files once the definition has been extended. Since we want this distinction (to take advantage of the `missingFiles`) but we are using the same API to achieve both, we need to consider the possibility that at the upload step we are actually creating a new bundle instead of just extending it into itself (i.e. when the returned `bundleHash` is the same as the requested `bundleHash`).

Reply:
"removal of the tests" what do you mean here? i did not remove tests. maybe it was some specific commit, where i commented tests just temporarly